File: section3/ros2_ws/src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

### the name of the file can be different from the name of the node and can be different from the name of the executable

def main(args=None):
    rclpy.init(args=args)

    # create a node
    ## Node Name: py_test
    node = Node("py_test")
    node.get_logger().info("Hello ROS2")

    # pause program allowing node to run
    rclpy.spin(node)


    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: section3/ros2_ws/src/my_cpp_pkg/src/my_first_node.cpp
#include <rclcpp/rclcpp.hpp>

int main(int args, char **argv) {
    // Initialize the ROS 2 system
    rclcpp::init(args, argv);


    // Create a node
    auto node = std::make_shared<rclcpp::Node>("cpp_test");
    RCLCPP_INFO(node->get_logger(), "Hello Cpp Node");

    rclcpp::spin(node);


    // Shutdown the ROS 2 system
    rclcpp::shutdown();

    return 0;
}